### PR TITLE
[ENH] estimator serialization: user choice of `serialization_format`, support for `cloudpickle`

### LINKED ISSUE #5324: [BUG] `LSTMFCNClassifier` `load_model` error: object has no attribute 'model_'
**Describe the bug**
error : 'LSTMFCNClassifier' object has no attribute 'model_'

**To Reproduce**

```python
from sktime.classification.deep_learning.lstmfcn import LSTMFCNClassifier
from sktime.utils import mlflow_sktime  
from sktime.datasets import load_italy_power_demand
X_train, y_train = load_italy_power_demand(split="train", return_type="numpy3D")
X_new, _ = load_italy_power_demand(split="test", return_type="numpy3D")
classifier=LSTMFCNClassifier(verbose=2,n_epochs=5)
classifier.fit(X_train,y_train)
mlflow_sktime.save_model(sktime_model=classifier,path='classifier')
classifier=mlflow_sktime.load_model(model_uri='classifier') 
classifier.predict(X_new)
```

**Versions**
sktime 0.22.0

### PR DESCRIPTION
**Pull request by achieveordie**
#### Reference Issues/PRs

In reference to #5324

#### What does this implement/fix? Explain your changes.

- Add support for `serialization_format` (currently `pickle` and `cloudpickle` are supported) for all estimators.
- Update `sktime_mlflow` to also be able to save DL estimators.

#### Does your contribution introduce a new dependency? If yes, which one?

None

#### Did you add any tests for the change?

Yes, I've added tests for all the added functionalities, I'm open to suggestions for adding more tests.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, I've added myself and possibly others to the [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) file - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Comment by achieveordie** [comment]
The errors seem unrelated, rerunning it to confirm.

**Comment by achieveordie** [comment]
The current CI failure is due to a known issue #5026.

File: sktime/base/_base.py [hunk 1]
```diff
@@ -177,6 +182,12 @@ def save(self, path=None):
                 path="/home/stored/estimator" then a zip file `estimator.zip` will be
                 stored in `/home/stored/`.
 
+        serialization_format: str, default = "pickle"
+            Module to use for serialization.
+            The available options are present under
```
Comment (fkiraly) [inline sktime/base/_base.py:187]: I would either spell it out, or print the `SERIALIZATION_FORMATS` constant in the docstring.
There are only two anyway.

File: sktime/classification/kernel_based/tests/test_arsenal.py [hunk 2]
```diff
@@ -6,6 +6,12 @@
 from sktime.tests.test_switch import run_test_for_class
 
 
+# A reference to this issue is also present inside sktime/tests/_config.py,
+# and needs to be removed from `EXCLUDED_TESTS` upon resolution.
+@pytest.mark.skip(
```
Comment (fkiraly) [inline sktime/classification/kernel_based/tests/test_arsenal.py:11]: this is sporadic, we should not skip it

**Review by fkiraly — CHANGES_REQUESTED** [review]
Nice!

Small blocking things:

* I would spell the available formats out in the docstring directly. There are just two options.
* I would split this PR up in two, one with the serialization format, one for `mlflow` extension.

Non-blocking, out of scope for this PR but perhaps worth a look: @benjaminbluhm has continued to develop the `mlflow` plugin as part of `mlflavours`. I am not sure of its status, but perhaps you want a quick look whether adding this in one of the two or both places makes sense, whether we want to backport sth, or outsource sth to `mlflavours`.

File: sktime/classification/kernel_based/tests/test_arsenal.py (hunk 2 above)
Comment (achieveordie) [reply sktime/classification/kernel_based/tests/test_arsenal.py:11]: What's the general consensus on dealing with such sporadic errors?

File: sktime/base/_base.py (hunk 1 above)
Comment (achieveordie) [reply sktime/base/_base.py:187]: Using a variable in the docstring would make things trickier with `__doc__()`. I might as well spell them here again, thanks.

**Comment by achieveordie** [comment]
[`mlflavors`](https://github.com/ml-toolkits/mlflavors) seems to have been updated quite a while ago. Ideally, this change should be present in both places since if one uses `mlflavors` with sktime DL estimators, they will get the same error as #5324.

If @benjaminbluhm doesn't mind, I'll open a similar PR there as well.

**Comment by achieveordie** [comment]
I'll make a follow-up PR for mlflow changes after this has been merged.

**Comment by achieveordie** [comment]
Two errors are related to #5488 and one error is from a tacky HTTPS connection, all unrelated to this PR.

**Review by fkiraly — APPROVED** [review]
Thanks!

**Review by fkiraly — DISMISSED** [review]
Thanks, looks good to me now!

**Pull request merged**